sickrage/providers/GenericProvider.py

```diff
@@ -518,3 +519,18 @@ def _make_url(self, result):
 
     def _verify_download(self, file_name=None):  # pylint: disable=unused-argument,no-self-use
         return True
+
+    @property
+    def recent_results(self):
+        if sickbeard.provider_recent_results.get(self.get_id()):
+            return sickbeard.provider_recent_results.get(self.get_id())[::-1]
+        else:
+            return []
+
+    @recent_results.setter
+    def recent_results(self, items):
+        if not sickbeard.provider_recent_results.get(self.get_id()):
+            sickbeard.provider_recent_results.update({self.get_id(): []})
+        sickbeard.provider_recent_results[self.get_id()] += items
+        if items:
+            del sickbeard.provider_recent_results[self.get_id()][:len(sickbeard.provider_recent_results[self.get_id()]) - self.max_recent_items]
```

Comment:
There should be always at least 5 items int the list. Then 2 will be added, making it 7. Then del list[:2] will result in removing 2 from the top.